lib/options-manager.js

```diff
@@ -607,6 +635,58 @@ const getOptionGroups = async (files, options) => {
 	return optionGroups;
 };
 
+async function recursiveBuildTsConfig(tsConfig, tsConfigPath) {
+	tsConfig = tsConfigAbsPaths(tsConfig, tsConfigPath);
+
+	if (!tsConfig.extends || (typeof tsConfig.extends === 'string' && tsConfig.extends.includes('node_modules'))) {
+		return tsConfig;
+	}
+
+	// If any of the following are missing, then we need to look up the base config as it could apply
+	const basePath = path.isAbsolute(tsConfig.extends)
+		? tsConfig.extends
+		: path.resolve(path.dirname(tsConfigPath), tsConfig.extends);
+
+	const baseTsConfig = await readJson(basePath);
+
+	delete tsConfig.extends;
+
+	tsConfig = {
+		compilerOptions: {
+			...baseTsConfig.compilerOptions,
+			...tsConfig.compilerOptions,
+		},
+		...baseTsConfig,
+		...tsConfig,
+	};
+
+	return recursiveBuildTsConfig(tsConfig, basePath);
+}
+
+// Convert all include, files, and exclude to absolute paths
+// and or globs. This works because ts only allows simple glob subset
+const tsConfigAbsPaths = (tsConfig, tsConfigPath) => {
+	if (Array.isArray(tsConfig.files)) {
+		tsConfig.files = tsConfig.files.map(
+			f => path.isAbsolute(f) ? f : path.resolve(path.dirname(tsConfigPath), f),
+		);
+	}
+
+	if (Array.isArray(tsConfig.include)) {
+		tsConfig.include = tsConfig.include.map(
+			f => path.isAbsolute(f) ? f : path.resolve(path.dirname(tsConfigPath), f),
+		);
+	}
+
+	if (Array.isArray(tsConfig.exclude)) {
+		tsConfig.exclude = tsConfig.exclude.map(
+			f => path.isAbsolute(f) ? f : path.resolve(path.dirname(tsConfigPath), f),
```

Comment:
The `path.isAbsolute` check is moot. `path.resolve` already passes `f` through if absolute.